frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeaderButtonRow.tsx

```diff
@@ -0,0 +1,315 @@
+  isEmpty?: boolean;
+  isPublic?: boolean;
+} & DashboardRefreshPeriodControls &
+  DashboardFullscreenControls &
+  Pick<
+    EmbedThemeControls,
+    "hasNightModeToggle" | "isNightMode" | "onNightModeChange"
+  >;
+
+type HeaderButtonProps = {
+  isEditing: boolean;
+  canEdit: boolean;
+  hasModelActionsEnabled: boolean;
+  isAnalyticsDashboard: boolean;
+  dashboard: Dashboard;
+  canManageSubscriptions: boolean;
+  formInput: any;
+  isAdmin: boolean;
+} & DashboardHeaderButtonRowProps;
+
+type ButtonConfig = {
+  component: (props: HeaderButtonProps) => ReactNode;
+  enabled: (props: HeaderButtonProps) => boolean;
+};
+
+const buttonConfigs: Record<ButtonKey, ButtonConfig> = {
+  [BUTTON_CONFIG.ADD_QUESTION]: {
+    component: () => <AddQuestionButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_HEADING_OR_TEXT]: {
+    component: () => <AddHeadingOrTextButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_LINK_CARD]: {
+    component: () => <AddLinkCardButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_SECTION]: {
+    component: () => <AddSectionButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_TEMPORAL_UNIT]: {
+    component: () => <AddTemporalUnitButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_FILTER_PARAMETER]: {
+    component: () => <AddFilterParameterButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_ACTION_ELEMENT]: {
+    component: () => <AddActionElementButton />,
+    enabled: ({ isEditing, canEdit, hasModelActionsEnabled }) =>
+      isEditing && canEdit && hasModelActionsEnabled,
+  },
+  [BUTTON_CONFIG.EXTRA_EDIT_BUTTONS_MENU]: {
+    component: () => <ExtraEditButtonsMenu />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.COPY_ANALYTICS_DASHBOARD]: {
+    component: () => <CopyAnalyticsDashboardButton />,
+    enabled: ({ isAnalyticsDashboard }) => isAnalyticsDashboard,
+  },
+  [BUTTON_CONFIG.EDIT_DASHBOARD]: {
+    component: ({ onRefreshPeriodChange }) => (
+      <EditDashboardButton onRefreshPeriodChange={onRefreshPeriodChange} />
+    ),
+    enabled: ({ isFullscreen, isEditing, canEdit }) =>
+      !isFullscreen && !isEditing && canEdit,
+  },
+  [BUTTON_CONFIG.DASHBOARD_SUBSCRIPTION]: {
+    component: () => <DashboardSubscriptionButton />,
+    enabled: ({
+      dashboard,
+      canManageSubscriptions,
+      formInput,
+      isAdmin,
+      isEditing,
+      isFullscreen,
+    }) =>
+      shouldRenderSubscriptionButton({
+        dashboard,
+        canManageSubscriptions,
+        formInput,
+        isAdmin,
+        isEditing,
+        isFullscreen,
+      }),
+  },
+  [BUTTON_CONFIG.DASHBOARD_EMBED_ACTION]: {
+    component: () => <DashboardEmbedAction />,
+    enabled: ({ dashboard, isPublic, isEmpty }) =>
+      !isEmpty && !isPublic && dashboard && !dashboard?.archived,
```

Comment:
```suggestion
      !isEmpty && !isPublic && dashboard && !dashboard.archived,
```

At this point, the dashboard will not be `null` since we already check in the condition before.